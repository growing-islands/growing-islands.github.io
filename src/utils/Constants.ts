type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
