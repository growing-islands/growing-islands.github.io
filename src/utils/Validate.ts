/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
