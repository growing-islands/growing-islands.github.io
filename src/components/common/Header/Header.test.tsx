import { render } from '~tests/libs/utils'
import { Header } from '.'

describe('components/common/Header', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<Header />)

    expect(asFragment()).toMatchSnapshot()
  })
})
