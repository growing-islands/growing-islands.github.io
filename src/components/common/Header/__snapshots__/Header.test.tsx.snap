// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Header Snapshot 1`] = `
<DocumentFragment>
  <header
    class=""
    data-testid="component"
  >
    <div>
      <h1
        data-testid="logo"
      >
        <a>
          <svg
            aria-label="Yumelab"
            viewBox="0 0 71.47 17.45"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M45.15,14.55a2.72,2.72,0,0,0,1-.18,4.63,4.63,0,0,0,.9-.52l.43.64a2.43,2.43,0,0,1-.48.34,3.49,3.49,0,0,1-.59.25c-.22.06-.44.12-.67.16a4,4,0,0,1-.69.06,4.67,4.67,0,0,1-1.39-.2,3.13,3.13,0,0,1-1.08-.59,2.59,2.59,0,0,1-.68-.94,3.23,3.23,0,0,1-.24-1.29A3.3,3.3,0,0,1,41.88,11a2.66,2.66,0,0,1,.64-1,2.88,2.88,0,0,1,1-.58,3.87,3.87,0,0,1,1.26-.2A3.76,3.76,0,0,1,46,9.45,2.85,2.85,0,0,1,47,10a2.76,2.76,0,0,1,.91,2.09v.75h-4.6a2,2,0,0,0,.63,1.3A1.89,1.89,0,0,0,45.15,14.55ZM44.77,10a1.4,1.4,0,0,0-.58.12,1.11,1.11,0,0,0-.47.37,2.1,2.1,0,0,0-.33.67,3.47,3.47,0,0,0-.13,1h3a3.05,3.05,0,0,0-.13-1,1.92,1.92,0,0,0-.34-.67,1.26,1.26,0,0,0-.48-.39A1.41,1.41,0,0,0,44.77,10Z"
            />
            <path
              d="M63.88,11.52h0A2.22,2.22,0,0,0,62,9.36a5.85,5.85,0,0,0-3.45.38l.28.72a4.22,4.22,0,0,1,2.37-.38,1.31,1.31,0,0,1,1.16,1.44H60.86A4.15,4.15,0,0,0,58.7,12,1.49,1.49,0,0,0,58,13.3a1.46,1.46,0,0,0,.7,1.34,4.05,4.05,0,0,0,2.16.44h3V11.52Zm-1.51,2.82H60.85c-.83,0-1.24-.35-1.24-1s.41-1,1.24-1h1.52Z"
            />
            <path
              d="M23.21,7.53a1.06,1.06,0,0,0-1,.5l-3,4.79v2.26h1.57V12.82L24.3,7.53Z"
            />
            <path
              d="M29.65,9.47h-.84v4.87h-1.4A1.68,1.68,0,0,1,26.29,14a1.42,1.42,0,0,1-.39-1.12V9.47H24.39v3.4a1.92,1.92,0,0,0,.74,1.66,3.79,3.79,0,0,0,2.28.55h2.92V10.13A.64.64,0,0,0,29.65,9.47Z"
            />
            <path
              d="M53.19,14.28V8.2a.64.64,0,0,0-.67-.67h-.9v7.55h5.79v-.8Z"
            />
            <path
              d="M71.24,11.11a2.61,2.61,0,0,0-.66-.88,3,3,0,0,0-1-.57,4.8,4.8,0,0,0-1.36-.19H66.83V8.2a.64.64,0,0,0-.67-.67h-.84v7.55h2.86a4.36,4.36,0,0,0,1.35-.19,3.1,3.1,0,0,0,1-.56,2.65,2.65,0,0,0,.66-.89,2.86,2.86,0,0,0,.23-1.16A2.94,2.94,0,0,0,71.24,11.11Zm-1.5,2.11a1.86,1.86,0,0,1-.35.64,1.2,1.2,0,0,1-.53.36,1.81,1.81,0,0,1-.67.12H66.83V10.22h1.36a1.81,1.81,0,0,1,.67.11,1.39,1.39,0,0,1,.53.36,1.86,1.86,0,0,1,.35.64,3.28,3.28,0,0,1,.12.95A3.21,3.21,0,0,1,69.74,13.22Z"
            />
            <path
              d="M39.73,10a3.79,3.79,0,0,0-2.28-.56H31.83v5.61h1.51V10.22h2v4.86h1.52V10.22h.54a1.63,1.63,0,0,1,1.12.34A1.42,1.42,0,0,1,39,11.68v2.74a.63.63,0,0,0,.67.66h.84v-3.4A1.92,1.92,0,0,0,39.73,10Z"
            />
            <path
              d="M19.22,12.82A3.47,3.47,0,0,0,18.54,9c-.81-.9-1.72-1.09-2.75-2.14a4.78,4.78,0,0,0,1.06,3C18,11,19,11.93,19.22,12.82Z"
            />
            <g>
              <path
                d="M15.63,2a2,2,0,1,1-2-2A2,2,0,0,1,15.63,2Z"
              />
              <path
                d="M11.54,7l.32.27a1.73,1.73,0,0,0,.48-1.21,1.78,1.78,0,0,0-1.78-1.78A1.8,1.8,0,0,0,9,5.22l.2.14Z"
              />
              <path
                d="M11.86,7.3a1.75,1.75,0,0,1-1.3.57A1.78,1.78,0,0,1,8.78,6.09,1.81,1.81,0,0,1,9,5.22L6.87,3.69A2.62,2.62,0,0,0,4.1,3.42a2.65,2.65,0,0,0-.93.8,3.2,3.2,0,0,0-.55,1.84s0,3.59,0,4.53L.88,13A1,1,0,0,0,0,14a1.11,1.11,0,0,0,.48.9L3,16.71a1.07,1.07,0,0,0,1.5-.19,1,1,0,0,0,.15-.78l1.79-2.5,4.21-1.42h0a3.17,3.17,0,0,0,1.57-1.12,2.55,2.55,0,0,0-.36-3.4Zm-1.43,3.87L6,12.66l-2.15,3a.38.38,0,0,1,.08.46.38.38,0,0,1-.53,0L.88,14.34a.38.38,0,0,1-.14-.51.37.37,0,0,1,.46-.07l2.12-2.95V9.94H5c.43,0,.63-.4.22-.79a1.42,1.42,0,0,1,.94-2.49A1.43,1.43,0,0,1,7.54,8.08a1.41,1.41,0,0,1-.48,1.07c-.42.39-.21.79.22.79h4.58A2.32,2.32,0,0,1,10.43,11.17Z"
              />
            </g>
            <ellipse
              cx="7.1"
              cy="16.93"
              rx="4.43"
              ry="0.52"
            />
          </svg>
        </a>
      </h1>
      <div>
        <nav
          class=""
        >
          <ul>
            <li>
              <a
                class="tablet:hidden"
              >
                <span>
                  HOME
                </span>
              </a>
            </li>
            <li>
              <a
                class=""
              >
                <span>
                  SERVICE
                </span>
                <span
                  class="tablet:hidden"
                >
                  サービス
                </span>
                <svg
                  aria-label="角"
                  class="hidden tablet:inline-block"
                  viewBox="0 0 21 31"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M.405.98h4.427l15.342 14.815L4.832 30.611H.527l15.347-14.816L.405.98Z"
                  />
                </svg>
              </a>
              <ul>
                <li>
                  <a
                    class=""
                  >
                    <span>
                      共創する受託開発
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    class=""
                  >
                    <span>
                      ユメラボのSES
                    </span>
                  </a>
                </li>
              </ul>
            </li>
            <li>
              <a
                class=""
              >
                <span>
                  MISSION
                </span>
                <span
                  class="tablet:hidden"
                >
                  ミッション
                </span>
              </a>
            </li>
            <li>
              <a
                class=""
              >
                <span>
                  COMPANY
                </span>
                <span
                  class="tablet:hidden"
                >
                  会社概要
                </span>
              </a>
            </li>
            <li>
              <a>
                <svg
                  aria-label="メール"
                  viewBox="0 0 512 384"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M464,64H48A48,48,0,0,0,0,112V400a48,48,0,0,0,48,48H464a48,48,0,0,0,48-48V112A48,48,0,0,0,464,64Zm0,48v40.8c-22.42,18.26-58.17,46.66-134.59,106.49C312.57,272.54,279.21,304.37,256,304c-23.21.37-56.58-31.46-73.41-44.71C106.18,199.46,70.43,171.07,48,152.8V112ZM48,400V214.4c22.91,18.25,55.41,43.86,104.94,82.64,21.85,17.21,60.13,55.19,103.06,55,42.72.23,80.51-37.2,103.05-54.95,49.53-38.78,82-64.4,105-82.65V400Z"
                    transform="translate(0 -64)"
                  />
                </svg>
                <span>
                  CONTACT
                </span>
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="tablet:hidden"
        >
          <span />
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
